openeo/api/logs.py

```diff
@@ -1,40 +1,34 @@
-class LogEntry:
-    """
-    A log entry.
-    """
-
-    def __init__(self, entry):
-        # Unique ID for the log, string, REQUIRED
-        self.id = entry['id']
-
-        # Error code, string, optional
-        self.code = entry.get("code", "")
-
-        # Severity level, string (error, warning, info or debug), REQUIRED
-        self.level = entry['level']
-
-        # Error message, string, REQUIRED
-        self.message = entry['message']
-
-        # Date and time of the error event as RFC3339 date-time, string, available since API 1.1.0
+"""
+LogEntry - log messages for jobs and services
+
+Properties:
+- id: Unique ID for the log, string, REQUIRED
+- code: Error code, string, optional
+- level: Severity level, string (error, warning, info or debug), REQUIRED
+- message: Error message, string, REQUIRED
+- time: Date and time of the error event as RFC3339 date-time, string, available since API 1.1.0
+- path: A "stack trace" for the process, array of dicts
+- links: Related links, array of dicts
+- usage: Usage metrics available as property 'usage', dict, available since API 1.1.0
+  May contain the following metrics: cpu, memory, duration, network, disk, storage and other custom ones
+  Each of the metrics is also a dict with the following parts: value (numeric) and unit (string)
+- data: Arbritrary data the user wants to "log" for debugging purposes.
+  Please note that this property may not exist as there's a difference
+  between None and non-existing. None for example refers to no-data in
+  many cases while the absence of the property means that the user did 
+  not provide any data for debugging.
+"""
+
+class LogEntry(dict):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Check required fields
+        missing = self._required.difference(self.keys())
```

Comment:
`_required` is not defined, right?

Reply:
Yes, forgot to copy, fixed.

Reply:
that's why the tests fail FYI

Reply:
Yes, forgot to copy, fixed.

Reply:
Still failing, but don't have access to the Jenkins...

Reply:
latest failures re due to github connection issue, the tests themselves didn't even run. Not sure if it just a temporary glitch or something serious

Reply:
Figured it out, should work now.

Reply:
tests run for me locally as well, so that's no blocker

Our s-ops team is looking at broken jenkins builds (it's a more serious problem apparently), I have no idea how long that will take

Reply:
oh sorry I spoke to soon, there are two actual test failures I didn't notice. 
I'll try to fix them

Reply:
fixed now, all tests pass on my setup